src/hackney_request.erl

```diff
@@ -81,6 +84,13 @@ perform(Client0, {Method0, Path, Headers0, Body0}) ->
                                               handle_body(Headers2, ReqType0, Body0, Client0)
```

Comment:
Here is the method you are bypassing when headers are disabled. Not sure what it does or what horrific things may happen to multipart streams if it isn't run. This you should definitely check 😈